### PR TITLE
chore: improve AuthenticationContext javadoc

### PR DESCRIPTION
**Pull request by mcollovati**

### CODE REVIEW

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/security/AuthenticationContext.java [hunk 1]
```diff
@@ -227,9 +244,15 @@ public boolean hasAnyRole(String... roles) {
 
     /**
      * Checks whether the current authenticated user has all the given roles.
+     * <p>
+     * </p>
+     * Roles must be provided without the role prefix, for example
+     * {@code hasAllRoles("USER", "ADMIN")} instead of
+     * {@code hasAllRoles("ROLE_USER", "ROLE_ADMIN)}.
```
Comment (tepi) [inline vaadin-spring/src/main/java/com/vaadin/flow/spring/security/AuthenticationContext.java:251]: Example does not match method parameter (collection)

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/security/AuthenticationContext.java [hunk 2]
```diff
@@ -195,9 +200,15 @@ public boolean hasRole(String role) {
 
     /**
      * Checks whether the current authenticated user has any of the given roles.
+     * <p>
+     * </p>
+     * Roles must be provided without the role prefix, for example
+     * {@code hasAnyRole("USER", "ADMIN")} instead of
+     * {@code hasAnyRole("ROLE_USER", "ROLE_ADMIN)}.
```
Comment (tepi) [inline vaadin-spring/src/main/java/com/vaadin/flow/spring/security/AuthenticationContext.java:207]: Example does not match method parameter (collection)

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/security/AuthenticationContext.java (hunk 1 above)
Comment (mcollovati) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/security/AuthenticationContext.java]: fixed

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/security/AuthenticationContext.java (hunk 2 above)
Comment (mcollovati) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/security/AuthenticationContext.java]: fixed

**Review by tepi — APPROVED** [review]

**Pull request merged**